src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs

```diff
@@ -213,6 +237,74 @@ public WatchViewModel(string label, string path, Action<string> tagGeometry, boo
             isCollection = label == WatchViewModel.LIST || label == WatchViewModel.DICTIONARY;
         }
 
+        private static string GetStringFromObject(object obj)
+        {
+            TypeCode type = Type.GetTypeCode(obj.GetType());
+            switch (type)
+            {
+                case TypeCode.Boolean:
+                    return ToString(obj);
+
+                case TypeCode.Double:
+                    return ((double)obj).ToString(numberFormat, CultureInfo.InvariantCulture);
+
+                case TypeCode.Int64:
+                    return ((long)obj).ToString(CultureInfo.InvariantCulture);
+
+                case TypeCode.DateTime:
+                    return ((DateTime)obj).ToString(PreferenceSettings.DefaultDateFormat, CultureInfo.InvariantCulture);
+
+                case TypeCode.Object:
+                    return ToString(obj);
+
+                default:
+                    return (string)obj;
+            };
+        }
+
+        private static string ToString(object obj)
+        {
+            return ReferenceEquals(obj, null) ?
+                Resources.NullString
```

Comment:
now this... looks confusing - please chop this up so it's more maintainable.

Also reference equals with `null`? Maybe just == `null`? What do you need to accomplish by using ReferenceEquals method - `null` is not a type so this line doesn't make much sense to me...

Reply:
see:
https://stackoverflow.com/questions/17667248/is-null-in-c-sharp-value-type-or-reference-type

Reply:
Actually just moved this from the `IWatchHandler` to the `WatchViewModel`  as we are now converting the object to a label string here. 

But you're right it is very confusing. Will have a look at it.

Reply:
@SHKnudsen can you use git blame to find out who wrote that code originally?